bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -319,4 +361,127 @@ private static void startHttpClient(HttpClient client) {
+                    response.getStatus(), length, mediaType, url);
+
+            if (response.getStatus() != HttpStatus.OK_200 || length == 0) {
+                logger.debug("Media download failed: unexpected return code {} (URL {})", response.getStatus(), url);
+                return null;
+            }
+
+            if (maxContentLength >= 0 && length > maxContentLength) {
+                logger.debug("Media download aborted: content length {} too big (URL {})", length, url);
+                return null;
+            }
+
+            String contentType = mediaType;
+            if (contentTypeRegex != null) {
+                if ((contentType == null || contentType.isEmpty()) && scanTypeInContent) {
+                    // We try to get the type from the data
+                    contentType = guessContentTypeFromData(data);
+                    logger.debug("Media download: content type from data: {} (URL {})", contentType, url);
+                }
+                if (contentType != null && contentType.isEmpty()) {
+                    contentType = null;
+                }
+                if (contentType == null) {
+                    logger.debug("Media download aborted: unknown content type (URL {})", url);
+                    return null;
+                } else if (!contentType.matches(contentTypeRegex)) {
+                    logger.debug("Media download aborted: unexpected content type \"{}\" (URL {})", contentType, url);
+                    return null;
+                }
+            } else if (contentType == null || contentType.isEmpty()) {
+                contentType = RawType.DEFAULT_MIME_TYPE;
+            }
+
+            rawData = new RawType(data, contentType);
+
+        } catch (IOException e) {
+            rawData = null;
+        }
+        if (rawData == null) {
+            logger.debug("Media download failed (URL {})", url);
+        } else {
+            logger.debug("Media downloaded: size {} type {} (URL {})", rawData.getBytes().length, rawData.getMimeType(),
+                    url);
+        }
+        return rawData;
+    }
+
+    /**
+     * Determine the content type from the data.
+     *
+     * @param data the data as a buffer of bytes
+     *
+     * @return the MIME type of the content, null if the content type could not be found
+     */
+    public static String guessContentTypeFromData(byte[] data) {
+        String contentType = null;
+        ByteArrayInputStream input = null;
+        try {
+            input = new ByteArrayInputStream(data);
+            contentType = URLConnection.guessContentTypeFromStream(input);
+            if (contentType != null && contentType.isEmpty()) {
+                contentType = null;
+            }
+        } catch (IOException e) {
+            logger.debug("Failed to determine content type: {}", e.getMessage());
+            contentType = null;
+        } finally {
+            IOUtils.closeQuietly(input);
```

Comment:
I assume you know that a resource is closed automatically by using try-with-resources.
In this case the `close` method of the `ByteArrayInputStream` is allowed to throw an `IOException` that need to be catch. So, you need an additional catch block, too (I assume you pointed to that topic -- correct?). But there is nothing we can do if the close fails at all (or are there any suggestions).

So, perhaps @kaikreuzer wanted to use a similar code as the following one, to get rid of the dependency.

```Java
    /**
     * Determine the content type from the data.
     *
     * @param data the data as a buffer of bytes
     * @return the MIME type of the content, null if the content type could not be found
     */
    public static String guessContentTypeFromData(byte[] data) {
        String contentType = null;
        try (final ByteArrayInputStream input = new ByteArrayInputStream(data)) {
            try {
                contentType = URLConnection.guessContentTypeFromStream(input);
                if (contentType != null && contentType.isEmpty()) {
                    contentType = null;
                }
            } catch (final IOException e) {
                logger.debug("Failed to determine content type: {}", e.getMessage());
            }
        } catch (final IOException ex) {
            // Error on closing input stream -- nothing we can do here.
        }
        return contentType;
    }
```